test/check-machines-create

```diff
@@ -1217,10 +1217,11 @@ vnc_password= "{vnc_passwd}"
                 b.click(".pf-c-modal-box__footer button:contains(Import)")
             else:
                 b.click(".pf-c-modal-box__footer button:contains(Create)")
-            init_state = "Creating VM installation" if dialog.start_vm else "Creating VM"
-            second_state = "Running" if dialog.start_vm else "Shut off"
 
-            self._assertVmStates(dialog.name, init_state, second_state)
+            if dialog.start_vm:
+                b.wait_text(f"#vm-{dialog.name}-state", "Running")
+            else:
+                self._assertVmStates(dialog.name, "Creating VM", "Shut off")
             b.wait_not_present("#create-vm-dialog")
 
         def _tryCreate(self, dialog):
```

Comment:
The implementation may have moved, but IMHO we should still check that this use case (changing settings before starting the VM) actually works. "We trust that this works" has bitten us way too often :grin:

Reply:
Alright, then at least I would change how we check that installation has started. [Here](https://github.com/cockpit-project/cockpit-machines/blob/main/test/check-machines-create#L1580) we check for that by looking if VM is persistent. Which was fine because before this PR, we undefined and then defined the VM (signal `Defined Added`), so for a short period the VM was not persistent.
But as I described in [this comment](https://github.com/cockpit-project/cockpit-machines/pull/583#discussion_r843704457), `virt-install --reinstall` doesn't undefine the VM, merely just updates it (signal `Defined Updated`), so the VM if persistent throughout the whole installation process, so the above-mentioned check will always be true.
Maybe we could remove that check, or instead listen for the signal `Defined Updated`.

Reply:
I sit()ed right before the `b.click("#vm-VmNotInstalled-install")`, and `virsh list --persistent` does not show anything. It only appears in `virsh list --all`, as it is not running (and `--persistent` only shows running VMs). I think the point of that check was to wait for the VM to start/run? So I'm not sure what you mean. Do you have time to rework the test in this PR to what you have in mind?